packages/virtual-core/src/index.ts

```diff
@@ -1010,8 +1001,8 @@ export class Virtualizer<
           const [latestOffset] = notUndefined(
             this.getOffsetForIndex(index, align),
           )
-
-          if (!approxEqual(latestOffset, this.getScrollOffset())) {
+          const currentScrollOffset = this.getScrollOffset()
+          if (!approxEqual(latestOffset, currentScrollOffset)) {
```

Comment:
Looks like there still a chance to get infinite loop

Reply:
Do you have something specific in mind?

Reply:
I guess for any given `scrollToIndex` we want to set some kind of counter, and break recursion when this counter is reach some limit, kinda dead loop detector